lib/css_parser/rule_set.rb

```diff
@@ -22,90 +24,152 @@ class RuleSet
       ['border-width', %w[border-top-width border-right-width border-bottom-width border-left-width]]
     ].freeze
 
-    class DeclarationValue
-      attr_reader :value
-      attr_accessor :important, :order
+    class Declarations
+      class Value
+        attr_reader :value
+        attr_accessor :important, :order
+
+        def initialize(value, important: nil, order: 0)
+          self.value = value
+          @important = important.nil? ? !value.match(CssParser::IMPORTANT_IN_PROPERTY_RX).nil? : important
+          @order = order
+        end
+
+        def value=(value)
+          value = value.to_s.sub(/\s*;\s*\Z/, '').gsub(CssParser::IMPORTANT_IN_PROPERTY_RX, '').strip
+          raise ArgumentError, 'value is empty' if value.empty?
+
+          @value = value.freeze
+        end
+
+        def to_s
+          return value unless important
 
-      def initialize(value, important: nil, order: 0)
-        self.value = value
-        @important = important.nil? ? !value.match(CssParser::IMPORTANT_IN_PROPERTY_RX).nil? : important
-        @order = order
+          "#{value} !important"
+        end
       end
 
-      def value=(value)
-        value = value.to_s.sub(/\s*;\s*\Z/, '').gsub(CssParser::IMPORTANT_IN_PROPERTY_RX, '').strip
-        raise ArgumentError, 'value is empty' if value.empty?
+      attr_reader :declarations
+      private :declarations
 
-        @value = value.freeze
+      def initialize(declarations = {})
+        @order = 0
+        @declarations = {}
+        declarations.each { |property, value| add_declaration!(property, value) }
       end
 
-      def to_s
-        return value unless important
+      # Add a CSS declaration
+      # @param [#to_s] property that should be added
+      # @param [Value, #to_s] value of the property
+      #
+      # @example
+      #   declarations['color'] = 'blue'
+      #
+      #   puts declarations['color']
+      #   => #<CssParser::RuleSet::Declarations::Value:0x000000000305c730 @important=false, @order=1, @value="blue">
+      #
+      # @example
+      #   declarations['margin'] = '0px auto !important'
+      #
+      #   puts declarations['margin']
+      #   => #<CssParser::RuleSet::Declarations::Value:0x00000000030c1838 @important=true, @order=2, @value="0px auto">
+      #
+      # If the property already exists its value will be over-written.
+      # If the value is empty - property will be deleted
+      def []=(property, value)
+        property = property.to_s.downcase
+        property.strip!
```

Comment:
dry up into normalize_property method ?

Reply:
Done, but I don't like it anyway, will take a look if we could normalize it just once in ruleset or wherever.